include/enrico/comm.h

```diff
@@ -140,6 +153,21 @@ class Comm {
   int rank = MPI_PROC_NULL;         //!< The calling process's rank in Comm:comm
 };
 
+template<typename T>
+void Comm::broadcast(std::vector<T>& values) const
+{
+  if (this->active()) {
+    // First broadcast the size of the vector
+    int n = values.size();
+    this->Bcast(&n, 1, MPI_INT);
+
+    // Resize vector (for rank != 0) and broacast data
+    if (values.size() != n)
+      values.resize(n);
+    this->Bcast(values.data(), n, get_mpi_type<T>());
```

Comment:
`size_t` and `ptrdiff_t` are typically typedefs to basic types, so as long as the basic types are covered we're good. On my system (gcc on x86_64), `size_t` is a typedef for `unsigned long` and `ptrdiff_t` is a typedef for `long`, and we have template specializations for both of those. I removed the default definition of `get_mpi_type<T>` so that we will get a compile-time error if we ever try to use it with a type that doesn't yet have a specialization.

Reply:
> I removed the default definition of get_mpi_type<T> so that we will get a compile-time error if we ever try to use it with a type that doesn't yet have a specialization.

Perfect.  Yes, I was concerned about that possibility.